packages/workshop-mcp/src/tools.ts

```diff
@@ -244,7 +244,16 @@ An error will be returned if no app is found for the given arguments.
 							stepNumber: nextProgress.stepNumber.toString(),
 							type: 'problem',
 						})
-						invariant(exerciseApp, 'No exercise app found')
```

Comment:
I prefer invariant here. The SDK will catch the error and return the content with the proper metadata. Please keep invariant for cases like this.